core/src/main/java/org/apache/iceberg/DefaultValueParser.java

```diff
@@ -0,0 +1,348 @@
+      case MAP:
+        Preconditions.checkArgument(
+            defaultValue.isObject() && defaultValue.has(KEYS) && defaultValue.has(VALUES) &&
+                defaultValue.get(KEYS).isArray() && defaultValue.get(VALUES).isArray(),
+            "Cannot parse %s to a %s value",
+            defaultValue, type);
+        JsonNode keys = defaultValue.get(KEYS);
+        JsonNode values = defaultValue.get(VALUES);
+        Preconditions.checkArgument(
+            keys.size() == values.size(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+
+        ImmutableMap.Builder<Object, Object> mapBuilder = ImmutableMap.builder();
+
+        Iterator<JsonNode> keyIter = keys.iterator();
+        Type keyType = type.asMapType().keyType();
+        Iterator<JsonNode> valueIter = values.iterator();
+        Type valueType = type.asMapType().valueType();
+
+        while (keyIter.hasNext()) {
+          mapBuilder.put(fromJson(keyType, keyIter.next()), fromJson(valueType, valueIter.next()));
+        }
+
+        return mapBuilder.build();
+      case STRUCT:
+        Preconditions.checkArgument(defaultValue.isObject(),
+            "Cannot parse default as a %s value: %s", type, defaultValue);
+        Types.StructType struct = type.asStructType();
+        StructLike defaultRecord = GenericRecord.create(struct);
+
+        List<Types.NestedField> fields = struct.fields();
+        for (int pos = 0; pos < fields.size(); pos += 1) {
+          Types.NestedField field = fields.get(pos);
+          String idString = String.valueOf(field.fieldId());
+          if (defaultValue.has(idString)) {
+            defaultRecord.set(pos, fromJson(field.type(), defaultValue.get(idString)));
+          }
+        }
+        return defaultRecord;
+      default:
+        throw new UnsupportedOperationException(String.format("Type: %s is not supported", type));
+    }
+  }
+
+  public static Object fromJson(Type type, String defaultValue) {
+    try {
+      JsonNode defaultValueJN = JsonUtil.mapper().readTree(defaultValue);
+      return fromJson(type, defaultValueJN);
+    } catch (IOException e) {
+      throw new IllegalArgumentException("Failed to parse: " + defaultValue + "; reason: " + e.getMessage(), e);
+    }
+  }
+
+  public static String toJson(Type type, Object defaultValue) {
+    return toJson(type, defaultValue, false);
+  }
+
+  public static String toJson(Type type, Object defaultValue, boolean pretty) {
+    try {
+      StringWriter writer = new StringWriter();
+      JsonGenerator generator = JsonUtil.factory().createGenerator(writer);
+      if (pretty) {
+        generator.useDefaultPrettyPrinter();
+      }
+      toJson(type, defaultValue, generator);
+      generator.flush();
+      return writer.toString();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @SuppressWarnings("checkstyle:CyclomaticComplexity")
```

Comment:
Same here. For new code, we should not suppress the complexity limit.

Reply:
seems I can remove the suppressing without causing any stylecheck issues.